quartz/plugins/emitters/contentPage.tsx

```diff
@@ -1,16 +1,52 @@
+import path from "path"
+import { visit } from "unist-util-visit"
+import { Root } from "hast"
+import { VFile } from "vfile"
 import { QuartzEmitterPlugin } from "../types"
 import { QuartzComponentProps } from "../../components/types"
 import HeaderConstructor from "../../components/Header"
 import BodyConstructor from "../../components/Body"
 import { pageResources, renderPage } from "../../components/renderPage"
 import { FullPageLayout } from "../../cfg"
-import { FilePath, joinSegments, pathToRoot } from "../../util/path"
+import { FilePath, isRelativeURL, joinSegments, pathToRoot } from "../../util/path"
 import { defaultContentPageLayout, sharedPageComponents } from "../../../quartz.layout"
 import { Content } from "../../components"
 import chalk from "chalk"
 import { write } from "./helpers"
 import DepGraph from "../../depgraph"
 
+// get all the dependencies for the markdown file
+// eg. images, scripts, stylesheets, transclusions
+const parseDependencies = (hast: Root, file: VFile): string[] => {
+  const dependencies: string[] = []
+
+  visit(hast, "element", (elem): void => {
+    let ref: string | null = null
+
+    if (["img", "script"].includes(elem.tagName) && elem?.properties?.src) {
+      ref = elem.properties.src.toString()
+    } else if (["a", "link"].includes(elem.tagName) && elem?.properties?.href) {
+      // transclusions will create a tags with relative hrefs
+      ref = elem.properties.href.toString()
+    }
+
+    // if it is a relative url, its a local file and we need to add
+    // it to the dependency graph. otherwise, ignore
+    if (ref === null || !isRelativeURL(ref)) {
+      return
+    }
+
+    let fp = path.join(file.data.filePath!, "..", ref).replace(/\\/g, "/")
```

Comment:
this is a very brittle way to normalize, `filePath` includes the `outdir` which can be any path on the system

instead of `file.data.filePath!`, use the normalized version

`const fp = path.posix.relative(argv.directory, file.data.filePath!)`

Reply:
also look at `quartz/plugins/transformers/links.ts` for some other link handling logic (could be good to pull some of these out into their own functions)

Reply:
`fp = path.posix.relative(argv.directory, file.data.filePath!)` wasn't returning the right thing, changed to `path.join(file.data.filePath!, path.relative(argv.directory, ref))`, lmk if that works